### PR TITLE
Tiled plotting for raw muon data

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
This is to make it easier for users to quickly inspect their raw data for unusual behaviour (e.g. broken or faulty detectors). 

**To test:**

Test fitting and plotting in muon analysis in general

To test the raw plots:
Load some data into muon analysis
In the plot window change the plot to "Raw Plot"
You will now see a plot of counts
You can turn it into a tiled plot

Check the above works for single and multi period.

Partial fix  for #29887 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
Once the reviewer is happy with the functionality I will edit the code, so that the pane does not appear in nightly.

**Comment by AnthonyLim23** [comment]
It can be slow for multiple runs (e.g. 11 runs). In the next stage I will add 2 selectors; 1 for the detectors and 1 for the run.

The speed of plotting one run is not that bad

**Review by sf1919 — CHANGES_REQUESTED** [review]
I've tested this with single and multi period data. For the most part is works well but I have observed a couple of issues. These may or may not be worth fixing.

The first is when you change the plot to raw data you get a legend that covers a large proportion of the left hand side of the plot. We don't want to get rid of the legend and it will be in the way no matter where it is but would it make more sense for it to default to the right? I'm assuming that most people will be looking at the plots from left to right and so get to see most of the plot before they need to move the legend.

Screenshot (4)

A small and strange glitch - when I loaded the latest MUSR run, changed to raw data and then clicked tile plot the tiled plots y axis ran from around 0 to 500. I then used the drop down menu and selected one of the detectors there didn't appear to be any changes to the plots. However when I changed the drop down to 'All' the y axis on the plots defaulted to around 0 to 200. Slightly strange - is that a bug or does it not matter because it can be changed manually if the user desires?

Finally if you leave the settings as Raw Data and the Tiled plot ticked and then go to change the instrument in the Home tab it retains these rather than default back to Plot Data and Tiled Plot not ticked. Is that the functionality we want?

I haven't looked over the code yet because I was confused by your message about testing functonality and then you'd change the code. Can you confirm whether or not I should look at the code now or when you have made the necessary changes.

**Comment by AnthonyLim23** [comment]
Yes, feel free to look at the code.

I am thinking of just having tiled plots on all the time by default (the single plot is not readable). That will fix the legend issue.

Maybe I should disable the ability to change from All, since the centre plots do not have their own axis.

I think it not changing with instrument is ok.

**Comment by sf1919** [comment]
I'll have a look at the code next.

In terms of your suggested changes:-

- Having tiled plots as default for raw data makes a lot of sense (I am assuming you mean it will only be default for raw data and not plot).
- Yes, if tiled plot ticked will disable the list of All/Detectors then that makes a lot of sense. I think you should make this change
- Less plots (less than 36) in the tiled view is a good idea as they are so small. My question is if you reduce it down to 25 (or would 20 make more sense??) how do you get the plots for detectors > 25 (or 20)?

As for not changing when the instrument changes thats ok but I will include a screen shot of what that looks like for reference.

Screenshot (5)

**Comment by AnthonyLim23** [comment]
The way the tiles work, its best to make it a whole number of rows and cols (so square number are great). Otherwise it (n-1)*n where n is the number of columns. 

I am going to disable this before it gets merged. In the next PR I will add the ability to allow users to change which detectors they are inspecting.

**Comment by AnthonyLim23** [comment]
I have reduced the number of spectra down to 16, it loads in an alright amount of time for multiperiod data (a single workspace)

**Comment by AnthonyLim23** [comment]
I found a bug when loading PSI data, I will do a fix for that today

**Comment by AnthonyLim23** [comment]
fixed it

**Review by sf1919 — CHANGES_REQUESTED** [review]
Thank you for making the raw data plot default to tiled view - I agree that deals with the legend issue very well. Also thank you for disabling the ability to choose specific plots - it really isn't necessary here.

I've run through the tests again for MUSR, EMU and PSI data and it all looks to be working well. Some good functionality here for plotting raw data. I have made some comments on the code though that may require some changes to be made.

File: scripts/Muon/GUI/Common/ADSHandler/workspace_naming.py
Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/ADSHandler/workspace_naming.py:39]: because I dont want the trailing under score

File: scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_utils.py
Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_utils.py:33]: probably, I can look at it

File: scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py
Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py:302]: is the plot condensed (no whitespace between subplots)

File: scripts/Muon/GUI/FrequencyDomainAnalysis/plot_widget/plot_freq_fit_pane_model.py
Comment (AnthonyLim23) [reply scripts/Muon/GUI/FrequencyDomainAnalysis/plot_widget/plot_freq_fit_pane_model.py:25]: good spot

File: scripts/Muon/GUI/MuonAnalysis/plot_widget/muon_analysis_plot_widget.py
Comment (AnthonyLim23) [reply scripts/Muon/GUI/MuonAnalysis/plot_widget/muon_analysis_plot_widget.py:68]: the main place those views are needed is in the corresponding presenter. The name in this file is not that important. I will have a think about a more clear way

Comment (sf1919) [reply scripts/Muon/GUI/MuonAnalysis/plot_widget/muon_analysis_plot_widget.py]: In the long run I definitely think we want to have clearer labels for these views so that its easier for others to work on this at a later date.

**Comment by AnthonyLim23** [comment]
looks like I cannot use `math.isqrt` due to RHEL7

File: scripts/Muon/GUI/MuonAnalysis/plot_widget/muon_analysis_plot_widget.py
Comment (AnthonyLim23) [reply scripts/Muon/GUI/MuonAnalysis/plot_widget/muon_analysis_plot_widget.py]: I have changed it

**Comment by AnthonyLim23** [comment]
I have fixed the failing test and I have commented out the code for adding raw plots to the GUI. So users will not see a difference

**Review by sf1919 — APPROVED** [review]

**Comment by sf1919** [comment]
I have tested the changes (including testing those lines now commented out so the functionality will not yet be available to users) and am happy with the changes. It works well and is a good addition to Muon Analysis plotting.

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we get this merged please? It is holding up other PR's

**Comment by DannyHindson** [comment]
Please can you add a release note?

**Comment by AnthonyLim23** [comment]
There is no need, because as far as the user is concerned there are no changes

**Pull request merged**
